core/src/main/java/io/knotx/forms/core/domain/FormsFactory.java

```diff
@@ -35,16 +39,57 @@ private FormsFactory() {
         .filter(f -> f.knots().stream().anyMatch(id -> id.startsWith(
             FormConstants.FRAGMENT_KNOT_PREFIX)))
         .map(f -> FormEntity.from(f, options))
+        .filter(Objects::nonNull)
         .collect(Collectors.toList());
-    if (areUnique(forms)) {
+
+    if (formIdentifiersAreNotUnique(forms)) {
```

Comment:
This detection can be moved after `toList()` in `FormsKnotProxy`.

Reply:
done